extensions/amp-analytics/0.1/test/test-requests.js

```diff
@@ -58,6 +58,103 @@ describes.realWin('Requests', {amp: 1}, env => {
   }
 
   describe('RequestHandler', () => {
+    describe('send with request origin', () => {
+      let spy;
+      beforeEach(() => {
+        spy = sandbox.spy();
+      });
+
+      it('should prepend request origin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': 'http://example.com'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r1');
+      });
+
+      it('handle trailing slash in request origin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': 'http://example.com/'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r1');
+      });
+
+      it('handle trailing path in request origin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': 'http://example.com/test'};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('http://example.com/r1');
+      });
+
+      it('handle empty requestOrigin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': ''};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('/r1');
+      });
+
+      it('handle undefined requestOrigin', function*() {
+        const r = {'baseUrl': '/r1', 'origin': undefined};
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith('/r1');
+      });
+
+      it('prepend request origin to absolute baseUrl', function*() {
+        const r = {
+          'baseUrl': 'https://baseurl.com',
+          'origin': 'https://requestorigin.com',
+        };
+        const handler = createRequestHandler(r, spy);
+        const expansionOptions = new ExpansionOptions({});
+
+        handler.send({}, {}, expansionOptions, {});
+        yield macroTask();
+        expect(spy).to.be.calledWith(
+          'https://requestorigin.comhttps://baseurl.com'
+        );
+      });
```

Comment:
Nice test coverage! Do you think it is worthing adding a test, where `baseUrl` doesn't have a leading slash?